### PR TITLE
[TR] PIM-4347: Change visibility of function getObjects in the BaseFetcher

### PR DESCRIPTION
**Pull request by ClementGautier**
The getObjects method here is usefull in other contexts than the BaseFetcher. Actually we need to use it in the EnterpriseEdition in the ProductDraftFetcher.

| Q                 | A
| ----------------- | ---
| Specs             | N/A
| Behats            | N/A
| Blue CI           | Y
| Changelog updated | N/A
| Review and 2 GTM  |  todo

### CODE REVIEW

**Comment by pierallard** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/base-fetcher.js [hunk 1]
```diff
@@ -132,6 +126,19 @@ define(['jquery', 'underscore', 'backbone', 'routing'], function ($, _, Backbone
...
                 this.entityPromises    = {};
             }
+        },
+
+        /**
+         * Wait for promises to resolve and return the promises results
+         *
+         * @param {Array|Object} promises
+         *
+         * @returns {Array}
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/base-fetcher.js]: It returns a Promise.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/base-fetcher.js [hunk 2]
```diff
@@ -132,6 +126,19 @@ define(['jquery', 'underscore', 'backbone', 'routing'], function ($, _, Backbone
                 this.entityListPromise = null;
                 this.entityPromises    = {};
             }
+        },
+
+        /**
+         * Wait for promises to resolve and return the promises results
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/fetcher/base-fetcher.js]: It does not retrun the promises results, it's confusing. It returns a promises that **will** return the results eventually.

**Comment by phaseinducer** [comment]
GTM

**Pull request merged**